### PR TITLE
support in-app sign-up

### LINKED ISSUE #162: App crashes when Signing Up in the app
## The steps to reproduce the bug from scratch:
1. fresh install, 5.7
2. at sign in / sign up, tap 'Create a Beeminder Account'
3. enter desired username, desired password, and email address
4. tap sign up
5. found nil while unwrapping

## A clear/concise description of what you expected to happen

Since offered, one can use the app to sign up for an account via email address.

## What actually happened instead

The account was created but the app did not proceed with it.

Instead

> Fatal error: Unexpectedly found nil while unwrapping an Optional value: 

# Notes

OAuthSignInManager.signUpWith(email
-> CurrentUserManager handleSuccessfulSignin

where a responseJSON["access_token"] is requested but does not exist in the response.

Instead the response contains an authentication_token and username
```
  "authentication_token" : "some auth token",
  "username" : "newly created account's username"
```

The difference between auth token and access token as mentioned https://api.beeminder.com/#auth.

### PR DESCRIPTION
**Pull request by krugerk**
replace accesstoken with token (apitoken) and support both authToken and accessToken

fixes #162

### CODE REVIEW

File: BeeSwift/CurrentUserManager.swift [hunk 1]
```diff
@@ -105,11 +118,25 @@ class CurrentUserManager : NSObject {
...
+        guard let authToken = responseJSON["authentication_token"].string, let username = responseJSON["username"].string else {
+            return
+        }
+        
+        CurrentUserManager.sharedManager.setApiToken(ApiToken(type: .AuthenticationToken, token: authToken))
+
+        UserDefaults.standard.set(username, forKey: usernameKey)
+        UserDefaults.standard.synchronize()
+        
+        NotificationCenter.default.post(name: Notification.Name(rawValue: CurrentUserManager.signedInNotificationName), object: self)
```
Comment (krugerk) [inline BeeSwift/CurrentUserManager.swift:131]: After successfully signing UP, what about next signing IN with that data?

The sign in response contains data about default notifications.

Comment (krugerk) [reply BeeSwift/CurrentUserManager.swift:131]: or fetching user, rather

**Comment by andrewpbrett** [comment]
This looks good, but could we use this as a first opportunity to write some failing tests and make them green? Just around the sign up/sign in flow for now.

**Comment by krugerk** [comment]
> This looks good, but could we use this as a first opportunity to write some failing tests and make them green? Just around the sign up/sign in flow for now. 

Been working on it - putting things in place to mock the server's responses.

**Review by andrewpbrett — CHANGES_REQUESTED** [review]
This looks good but we'd need to have tests for the various operations that use the API token before it can be merged.

**Pull request closed without merging**